modules/repository-url/src/test/java/org/elasticsearch/repositories/url/URLFixture.java

```diff
@@ -0,0 +1,160 @@
+            httpServer.createContext("/", new ResponseHandler(dir(args[1])));
+            httpServer.start();
+
+            // Wait to be killed
+            Thread.sleep(Long.MAX_VALUE);
+
+        } finally {
+            httpServer.stop(0);
+        }
+    }
+
+    @SuppressForbidden(reason = "Paths#get is fine - we don't have environment here")
+    private static Path dir(final String dir) {
+        return Paths.get(dir);
+    }
+
+    private static void writeFile(final Path dir, final String fileName, final String content) throws IOException {
+        final Path tempPidFile = Files.createTempFile(dir, null, null);
+        Files.write(tempPidFile, singleton(content));
+        Files.move(tempPidFile, dir.resolve(fileName), StandardCopyOption.ATOMIC_MOVE);
+    }
+
+    private static String addressToString(final SocketAddress address) {
+        final InetSocketAddress inetSocketAddress = (InetSocketAddress) address;
+        if (inetSocketAddress.getAddress() instanceof Inet6Address) {
+            return "[" + inetSocketAddress.getHostString() + "]:" + inetSocketAddress.getPort();
+        } else {
+            return inetSocketAddress.getHostString() + ":" + inetSocketAddress.getPort();
+        }
+    }
+
+    static class ResponseHandler implements HttpHandler {
+
+        private final Path repositoryDir;
+
+        ResponseHandler(final Path repositoryDir) {
+            this.repositoryDir = repositoryDir;
+        }
+
+        @Override
+        public void handle(HttpExchange exchange) throws IOException {
+            Response response;
+            if ("GET".equalsIgnoreCase(exchange.getRequestMethod())) {
+                String path = exchange.getRequestURI().toString();
+                if (path.length() > 0 && path.charAt(0) == '/') {
+                    path = path.substring(1);
+                }
+
+                Path normalizedRepositoryDir = repositoryDir.normalize();
+                Path normalizedPath = normalizedRepositoryDir.resolve(path).normalize();
+
+                if (normalizedPath.startsWith(normalizedRepositoryDir)) {
+                    if (Files.exists(normalizedPath) && Files.isReadable(normalizedPath) && Files.isRegularFile(normalizedPath)) {
+                        byte[] content = Files.readAllBytes(normalizedPath);
+                        Map<String, String> headers = singletonMap("Content-Length", String.valueOf(content.length));
+                        response = new Response(RestStatus.OK, headers, "application/octet-stream", content);
+                    } else {
+                        response = new Response(RestStatus.NOT_FOUND, emptyMap(), "text/plain", new byte[0]);
+                    }
+                } else {
+                    response = new Response(RestStatus.FORBIDDEN, emptyMap(), "text/plain", new byte[0]);
+                }
+            } else {
+                response = new Response(RestStatus.INTERNAL_SERVER_ERROR, emptyMap(), "text/plain", new byte[0]);
```

Comment:
No "unsupported HTTP method" message? :)

Reply:
That's because I'm lazy :) I added the message.